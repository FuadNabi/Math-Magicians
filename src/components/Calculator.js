import React from 'react';
import calculate from '../logic/calculate';

export default class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.buttonClick = this.buttonClick.bind(this);
  }

  buttonClick(element) {
    this.setState((state) => calculate(state, element.target.textContent));
  }

  render() {
    const { next } = this.state;
    const { total } = this.state;

    return (
      <div className="buttons-cont">
        <div className="calc-section">{next || total || 0}</div>
        <div className="grid-container">
          <div className="container-row">
            <button type="button" onClick={this.buttonClick} className="button-grey">AC</button>
            <button type="button" onClick={this.buttonClick} className="button-grey">+/-</button>
            <button type="button" onClick={this.buttonClick} className="button-grey">%</button>
            <button type="button" onClick={this.buttonClick} className="button-orange">รท</button>
          </div>

          <div className="container-row">
            <button type="button" onClick={this.buttonClick} className="button-grey">7</button>
            <button type="button" onClick={this.buttonClick} className="button-grey">8</button>
            <button type="button" onClick={this.buttonClick} className="button-grey">9</button>
            <button type="button" onClick={this.buttonClick} className="button-orange">x</button>
          </div>

          <div className="container-row">
            <button type="button" onClick={this.buttonClick} className="button-grey">4</button>
            <button type="button" onClick={this.buttonClick} className="button-grey">5</button>
            <button type="button" onClick={this.buttonClick} className="button-grey">6</button>
            <button type="button" onClick={this.buttonClick} className="button-orange">-</button>
          </div>

          <div className="container-row">
            <button type="button" onClick={this.buttonClick} className="button-grey">1</button>
            <button type="button" onClick={this.buttonClick} className="button-grey">2</button>
            <button type="button" onClick={this.buttonClick} className="button-grey">3</button>
            <button type="button" onClick={this.buttonClick} className="button-orange">+</button>
          </div>

          <div className="third-row-bottom">
            <button type="button" onClick={this.buttonClick} className="button-grey">0</button>
            <button type="button" onClick={this.buttonClick} className="button-grey">.</button>
            <button type="button" onClick={this.buttonClick} className="button-orange">=</button>
          </div>
        </div>

      </div>
    );
  }
}
