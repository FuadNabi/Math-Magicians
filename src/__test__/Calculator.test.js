import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { fireEvent } from '@testing-library/dom';
import Calculator from '../components/Calculator';

test('Calculator screen is equal to 0', () => {
  const tree = render(<Calculator />);
  const calculatorInput = tree.container.querySelector('.calc-section');
  expect(calculatorInput.innerHTML).toBe('0  ');
});

test('Calculator page test', () => {
  const tree = renderer.create(<Calculator />).toJSON;
  expect(tree).toMatchSnapshot();
});

test('Ther user type 2 + 2', () => {
  const tree = render(<Calculator />);
  const calculatorInput = tree.container.querySelector('.calc-section');
  let button = tree.getByText('2');
  fireEvent.click(button);
  expect(calculatorInput.innerHTML).toBe('  2');
  button = tree.getByText('+');
  fireEvent.click(button);
  expect(calculatorInput.innerHTML).toBe('2 + ');
  button = tree.getByText('2');
  fireEvent.click(button);
  expect(calculatorInput.innerHTML).toBe('2 + 2');
  button = tree.getByText('=');
  fireEvent.click(button);
  expect(calculatorInput.innerHTML).toBe('4  ');
});

test('Ther user tries devide by 0', () => {
  const tree = render(<Calculator />);
  const calculatorInput = tree.container.querySelector('.calc-section');
  let button = tree.getByText('2');
  fireEvent.click(button);
  expect(calculatorInput.innerHTML).toBe('  2');
  button = tree.getByText('รท');
  fireEvent.click(button);
  expect(calculatorInput.innerHTML).toBe('2 รท ');
  button = tree.getByText('0');
  fireEvent.click(button);
  expect(calculatorInput.innerHTML).toBe('2 รท 0');
  button = tree.getByText('=');
  fireEvent.click(button);
  expect(calculatorInput.innerHTML).toBe('Can\'t divide by 0.  ');
});
