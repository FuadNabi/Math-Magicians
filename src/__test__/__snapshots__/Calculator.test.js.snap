// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator page test 1`] = `[Function]`;
